### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by pyramation**

### CODE REVIEW

File: src/pages/postgraphile/postgresql-indexes.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ WITH indexed_tables AS (
       i.relname as index_name,
       array_to_string(array_agg(a.attname), ', ') as column_names,
       ix.indrelid,
-      ix.indkey
+      string_to_array(ix.indkey::text, ' ')::smallint[] as indkey
```
Comment (benjie) [inline src/pages/postgraphile/postgresql-indexes.md]: Casting directly to `::smallint[]` seems safe:

```
[mindtree] # select indkey::smallint[] from pg_index;
      indkey
------------------
 [0:1]={1,2}
 [0:0]={1}
 [0:0]={1}
 [0:0]={-2}
 [0:3]={1,2,3,4}
```

File: src/pages/postgraphile/postgresql-indexes.md [hunk 2]
```diff
@@ -74,11 +74,11 @@ AND NOT EXISTS(
   SELECT 1
   FROM indexed_tables
   WHERE indrelid = conrelid
-  AND conkey::smallint[] = indkey::smallint[]
-  OR (array_length(indkey, 1) > 1 AND indkey::smallint[] @> conkey)
+  AND conkey = indkey
+  OR (array_length(indkey, 1) > 1 AND indkey @> conkey)
```
Comment (benjie) [inline src/pages/postgraphile/postgresql-indexes.md]: This is tidier - I like the cast above to enable this 👍

**Comment by benjie** [comment]
(Incidentally the currently published code works just fine for me on PG 10.4, but the medium post does not.)

**Comment by pyramation** [comment]
I can confirm changing to `indkey::smallint[]` does not work for me.

**Comment by pyramation** [comment]
Specifically, here is where I found proper cast: https://stackoverflow.com/questions/28072583/how-can-i-test-if-int2vector-contains-exactly-one-specific-value

**Comment by benjie** [comment]
Why though... is this a localisation issue? Did it change in 10.4 vs 10.3? Why does it work for me but not you. This is troubling.

**Comment by pyramation** [comment]
Let me make an example! I'll put together a demo

**Comment by pyramation** [comment]
Ok, so here is before the index:

screen shot 2018-09-29 at 11 01 46 am

They all fail, which means they all detect there is no index. This is good!

https://travis-ci.org/pyramation/detect-indexes-example-test/jobs/435045250

and here is after the index:

screen shot 2018-09-29 at 11 02 00 am

https://travis-ci.org/pyramation/detect-indexes-example-test/jobs/435045251

Here, only the updated one detects that I actually added the index.

I'm down to try more versions of postgres, currently this is using 10.3

**Comment by pyramation** [comment]
I just tested out `postgres:10.5`, also seems to have the exact same results

https://travis-ci.org/pyramation/detect-indexes-example-test/builds/435052306

**Comment by pyramation** [comment]
here is a matrix of 10.3, 10.4, 10.5 https://travis-ci.org/pyramation/detect-indexes-example-test/builds/435056588

**Comment by pyramation** [comment]
also here is a passing style version of the test:

https://travis-ci.org/pyramation/detect-indexes-example-test/builds/435060118

screen shot 2018-09-29 at 12 18 29 pm

screen shot 2018-09-29 at 12 18 35 pm

**Pull request merged**